elyra/pipeline/validate.py

```diff
@@ -0,0 +1,633 @@
+                                                           "nodeName": node_label,
+                                                           "propertyName": node_property})
+
+    def _validate_container_image_name(self, node_id: str, node_label: str, image_name: str,
+                                       response: ValidationResponse) -> None:
+        """
+        Validates the image name exists and is proper in syntax
+        :param node_id: the unique ID of the node
+        :param node_label: the given node name or user customized name/label of the node
+        :param response: ValidationResponse containing the issue list to be updated
+        """
+        if not image_name:
+            response.add_message(severity=ValidationSeverity.Error,
+                                 message_type="invalidNodeProperty",
+                                 message="Node is missing image name",
+                                 data={"nodeID": node_id,
+                                       "nodeName": node_label,
+                                       "propertyName": 'runtime_image'})
+
+    def _validate_resource_value(self, node_id: str, node_label: str, resource_name: str,
+                                 resource_value: str, response: ValidationResponse) -> None:
+        """
+        Validates the value for hardware resources requested
+        :param node_id: the unique ID of the node
+        :param node_label: the given node name or user customized name/label of the node
+        :param response: ValidationResponse containing the issue list to be updated
+        :param resource_name: the name of the resource e.g. cpu, gpu. memory
+        :param resource_value: the value of the resource
+        """
+        try:
+            if int(resource_value) <= 0:
+                response.add_message(severity=ValidationSeverity.Error,
+                                     message_type="invalidNodeProperty",
+                                     message="Property must be greater than zero",
+                                     data={"nodeID": node_id,
+                                           "nodeName": node_label,
+                                           "propertyName": resource_name})
+        except (ValueError, TypeError):
+            response.add_message(severity=ValidationSeverity.Error,
+                                 message_type="invalidNodeProperty",
+                                 message="Property has a non-parsable value",
+                                 data={"nodeID": node_id,
+                                       "nodeName": node_label,
+                                       "propertyName": resource_name})
+
+    def _validate_filepath(self, node_id: str, node_label: str, property_name: str, server_root: str,
+                           filename: str, response: ValidationResponse, root_dir: Optional[str] = None) -> None:
+        """
```

Comment:
the "root" parameters are confusing.  I think think `server_root` should be removed and `self.root_dir` used for _that_ value, and the `root_dir` parameter should be renamed to what appears to be `file_dir` - wherever the file to run resides.

Reply:
fixed